### PR TITLE
JDBCToGCS - Added support to read SQL from GCS

### PR DESCRIPTION
**Pull request by vanshaj-bhatia**

### CODE REVIEW

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java [hunk 1]
```diff
@@ -64,16 +71,48 @@ public JDBCToGCS() {
...
+      LOGGER.error(
+          "Required parameter not passed. Either {} or {} parameter must be provided. ",
+          JDBC_TO_GCS_SQL,
+          JDBC_TO_GCS_SQL_FILE);
+      throw new IllegalArgumentException(
+          "Required parameters for JDBCToGCS not passed. "
+              + "Set mandatory parameter for JDBCToGCS template "
+              + "in resources/conf/template.properties file or at runtime. "
+              + "Refer to jdbc/README.md for more instructions.");
+    } else if (StringUtils.isAllBlank(jdbcSQL)) {
```
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:84]: The condition here is cryptic.
Instead please use: StringUtils.isNotBlank(jdbcSQLFile)

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java [hunk 2]
```diff
@@ -64,16 +71,48 @@ public JDBCToGCS() {
...
+      Matcher matches = Pattern.compile("gs://(.*?)/(.*)").matcher(jdbcSQLFile);
+      matches.matches();
+      String bucket = matches.group(1);
+      String objectPath = matches.group(2);
+
+      Storage storage = StorageOptions.getDefaultInstance().getService();
+      Blob blob = storage.get(bucket, objectPath);
+      String fileContent = new String(blob.getContent());
+      jdbcSQL = "(" + fileContent + ") as a";
+    } else if (StringUtils.isAllBlank(jdbcSQLFile)) {
```
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:94]: The condition here is cryptic.
Instead please use: StringUtils.isNotBlank(jdbcSQL)

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java (hunk 1 above)
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java:84]: code should also throw an error when both jdbcSQL and jdbcSQLFile are populated.
i.e. it should ensure only one is provided. Also refer to GCSToJDBCConfig.java if all these validations can be done more neatly (somehow, I am not sure).

**Review by shashank-google — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java (hunk 1 above)
Comment (vanshaj-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java]: Moved all validations from driver file (JDBCToGCS.java) to *Config.java file. Re-wrote all the validations with Jakarta bean validation. Please review.

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java (hunk 2 above)
Comment (vanshaj-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java]: Moved all validations from driver file (JDBCToGCS.java) to *Config.java file. Re-wrote all the validations with Jakarta bean validation. Please review.

File: java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java (hunk 1 above)
Comment (vanshaj-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java]: Added JDBCToGCSConfig.java. Please review.

Comment (vanshaj-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java]: @shashank-google Please review the changes.

**Comment by shashank-google** [comment]
@vanshaj-bhatia branch is out of date, needs to be updated with head

**Review by shashank-google — APPROVED** [review]

**Comment by shashank-google** [comment]
@vanshaj-bhatia I have approved the changss.
Please ensure that java build is not breaking from this cl. I feel you probably need to run, `mvn install` before checkin

**Pull request merged**